agent/src/beerocks/slave/gate/unit_test/gate_test.cpp

```diff
@@ -0,0 +1,121 @@
+*/
+
+const auto associated_sta_mac = sMacAddr{0x01, 0x02, 0x03, 0x04, 0x05, 0x06};
+const uint8_t operating_class = 0x73;
+const uint8_t channel_number  = 255;
+const uint8_t channel_1       = 46;
+const uint8_t channel_2       = 31;
+
+/**
+* helper function to fill 1905 values
+*/
+static bool set_1905_values(ieee1905_1::CmduMessageTx &cmdu1905)
+{
+    auto beacon_metrics_query = cmdu1905.getClass<wfa_map::tlvBeaconMetricsQuery>();
+    if (!beacon_metrics_query) {
+        return false;
+    }
+
+    // fill some values
+    beacon_metrics_query->associated_sta_mac() = associated_sta_mac;
+    beacon_metrics_query->operating_class()    = operating_class;
+    // for ap channel report list, this must be set to 255
+    beacon_metrics_query->channel_number() = channel_number;
+
+    auto ap_channel_list = beacon_metrics_query->create_ap_channel_reports_list();
+    if (!ap_channel_list) {
+        return false;
+    }
+
+    ap_channel_list->alloc_ap_channel_report_list(3);
+    // the first is operating class, which is currently ignored system wide
+    *ap_channel_list->ap_channel_report_list(0) = operating_class;
+    *ap_channel_list->ap_channel_report_list(1) = channel_1;
+    *ap_channel_list->ap_channel_report_list(2) = channel_2;
+
+    beacon_metrics_query->add_ap_channel_reports_list(ap_channel_list);
+
+    return true;
+}
+
+TEST(gate_beacon_query_test, loading_vs_specific_from_1905_beacon_query)
+{
+    // preperation for beerocks
+
+    // class list
+
+    constexpr const uint16_t BUF_SIZE_BEEROCKS = 10000; // just an arbitrary size
+    uint8_t buff_beerocks[BUF_SIZE_BEEROCKS];
+    ClassList cl(buff_beerocks, BUF_SIZE_BEEROCKS);
+
+    // beerocks
+    cl.addClass<beerocks_message::cACTION_MONITOR_CLIENT_BEACON_11K_REQUEST>();
+
+    auto beerocks = cl.getClass<beerocks_message::cACTION_MONITOR_CLIENT_BEACON_11K_REQUEST>();
+
+    ASSERT_NE(beerocks, nullptr);
+
+    // build 1905 message whthin a Tx message
+
+    // construct a Tx message with beacon metrics query
+    constexpr const uint16_t BUF_SIZE_1905_TX = 10000; // just an arbitrary size
+    uint8_t buff_1905_tx[BUF_SIZE_1905_TX];
+    ieee1905_1::CmduMessageTx cmdu_1905(buff_1905_tx, BUF_SIZE_1905_TX);
+    cmdu_1905.create(0x1, ieee1905_1::eMessageType::BEACON_METRICS_QUERY_MESSAGE);
+    cmdu_1905.addClass<wfa_map::tlvBeaconMetricsQuery>();
+
+    // fill 1905 message with values
+    bool filled = set_1905_values(cmdu_1905);
+    ASSERT_TRUE(filled);
+
+    // Rx - built from the tx message above
+    ieee1905_1::CmduMessageRx rx(cmdu_1905.getMessageBuff(), cmdu_1905.getMessageBuffLength());
+    rx.parse();
+    auto tlv_becon_metrics_query = rx.getClass<wfa_map::tlvBeaconMetricsQuery>();
+
+    ASSERT_NE(tlv_becon_metrics_query, nullptr);
+
+    // DUT -
+    beerocks::gate::load(/*dst*/ beerocks, /*src*/ rx);
+
+    auto &beerocks_params = beerocks->params();
+
+    // set
+    ASSERT_EQ(beerocks_params.channel, channel_number);
```

Comment:
Please consider using EXPECT_EQ instead of ASSERT_EQ
See [Assertions](https://github.com/google/googletest/blob/master/googletest/docs/primer.md#assertions)

Reply:
done.